src/csharp/Grpc.Microbenchmarks/UnaryCallOverheadBenchmark.cs

```diff
@@ -0,0 +1,93 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Grpc.Core;
+using Grpc.Core.Internal;
+using System;
+
+namespace Grpc.Microbenchmarks
+{
+    // this test creates a real server and client, measuring the inherent inbuilt
+    // platform overheads; the marshallers **DO NOT ALLOCATE**, so any allocations
+    // are from the framework, not the messages themselves
+
+    // important: allocs are not reliable on .NET Core until .NET Core 3, since
+    // this test involves multiple threads
+
+    [ClrJob, CoreJob] // test .NET Core and .NET Framework
+    [MemoryDiagnoser] // allocations
+    public class UnaryCallOverheadBenchmark
+    {
+        private static readonly Task<string> CompletedString = Task.FromResult("");
+        private static readonly byte[] EmptyBlob = new byte[0];
+        private static readonly Marshaller<string> EmptyMarshaller = new Marshaller<string>(_ => EmptyBlob, _ => "");
+        private static readonly Method<string, string> PingMethod = new Method<string, string>(MethodType.Unary, nameof(PingBenchmark), "Ping", EmptyMarshaller, EmptyMarshaller);
+
+        [Benchmark]
+        public string Ping()
+        {
+            return client.Ping("", new CallOptions());
+        }
+
+        Channel channel;
+        PingClient client;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            // create client
+            channel = new Channel("localhost", 10042, ChannelCredentials.Insecure);
```

Comment:
nit: suggest aiming this at a bogus port number, e.g. like port 0.

Perhaps hat or, a comment here to note how this Channel will never actually connect